catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/RoleRepository.java

```diff
@@ -340,8 +338,97 @@ public RoleUpdater(Role original, Role updated, Operation operation) {
     }
 
     @Override
-    public void entitySpecificUpdate() throws IOException {
-      recordChange("default", original.getEntity().getDefault(), updated.getEntity().getDefault());
+    public void entitySpecificUpdate() throws IOException, ParseException {
+      updateDefault(original.getEntity(), updated.getEntity());
+    }
+
+    private void updateDefault(Role origRole, Role updatedRole) throws IOException, ParseException {
+      long startTime = System.nanoTime();
+      if (Boolean.FALSE.equals(origRole.getDefault()) && Boolean.TRUE.equals(updatedRole.getDefault())) {
+        setDefaultToTrue(updatedRole);
+      }
+      if (Boolean.TRUE.equals(origRole.getDefault()) && Boolean.FALSE.equals(updatedRole.getDefault())) {
+        setDefaultToFalse(updatedRole);
+      }
+      recordChange("default", origRole.getDefault(), updatedRole.getDefault());
+      LOG.debug(
+          "Took {} ns to update {} role field default from {} to {}",
+          System.nanoTime() - startTime,
+          updatedRole.getName(),
+          origRole.getDefault(),
+          updatedRole.getDefault());
+    }
+
+    private void setDefaultToTrue(Role role) throws IOException, ParseException {
+      List<Role> defaultRoles = getDefaultRoles(null, ROLE_PATCH_FIELDS);
+      EntityRepository<Role> roleRepository = Entity.getEntityRepository(Entity.ROLE);
+      // Set default=FALSE for all existing default roles.
+      for (Role defaultRole : defaultRoles) {
```

Comment:
There will be only one default role here (at any given point of time) based on how this feature has been implemented.